src/lib.rs

```diff
@@ -471,25 +471,25 @@ impl<Types: IpfsTypes> Ipfs<Types> {
     /// Subscribes to a given topic. Can be done at most once without unsubscribing in the between.
     /// The subscription can be unsubscribed by dropping the stream or calling
     /// [`pubsub_unsubscribe`].
-    pub async fn pubsub_subscribe(&self, topic: &str) -> Result<SubscriptionStream, Error> {
+    pub async fn pubsub_subscribe(&self, topic: String) -> Result<SubscriptionStream, Error> {
```

Comment:
So basically the rule here for the `ipfs::Ipfs` API would be: take ownership of the value we know we'll have to have clone either way.

Reply:
I believe it is preferable to be upfront about the need to own the value when possible, instead of having concealed clones after providing a reference. It is unlikely to be any issue in this particular case, just a general rule that can't hurt.